students/CarolF/module03/MailRoom_1.py

```diff
@@ -0,0 +1,106 @@
+        print("The name entered:" ,donorName) ###REMOVE AFTER DEBUGGING
+    
+    if(donorName not in justDonors): 
+        print(donorName, "Is not in list. It will be added to donor list") #remove after debugging
+        isNewDonor = True
+        justDonors.append(donorName)
+        retrieveDonationAmount(donorName, isNewDonor)
+        
+    elif (donorName in justDonors):
+        print("Donor was found. You want to send a thank you to ", donorName)  #remove after debugging
+        isNewDonor = False
+        retrieveDonationAmount(donorName, isNewDonor)
+
+
+
+def retrieveDonationAmount(donorName, isNewDonor):
+    """requests donor information and will append new contibution to former donor"""
+    newContribution = float(input("please enter new contribution amount....> "))
+    if isNewDonor is False:
+        for donor in donorList:
+        	if donor[0] == donorName:
+        		donor.append(newContribution)
+        		printLetter(donorName,newContribution)
+    else:
+        donorList.append([donorName,newContribution]) 
+        printLetter(donorName,newContribution)
+
+    print(donorList)
+
+def printLetter(donorName, newContribution):
+    """completes and prints Thank you letter then calls getAction method to request new action"""
+    print("\nDear {} ,\n".format(donorName))
+    print("Thank you for your generous contribution of ${} ".format(newContribution))
+    print("We will try to spend it wisely. No company trips to the DQ, pinky swear.")
+    print("Sincerely,")
+    print("Us\n")
+    getAction()
+
+
+#still need to call and improve formattting.
+def createReportData():
+    print('{:<15}'.format("Donor Name"), '{:<20}'.format('|  Total Given  |'),
+                 '{:>5}'.format('Number of Gifts  |'),'{:>20}'.format('Average Gift  |')) 
+    print('_'*77)
+    dataForDonationReport = []
+    for donor  in donorList: # prepare the report list
+        donorNumTotal =sum(donor[1:])
+        numbGifts = (len(donor)-1)
+        donorAverage = "$ {0:.2f}".format(float((sum(donor[1:]))/numbGifts)) #subtract 1 to acocunt for name placeholder in List[0]
+        dataForDonationReport.append([donorNumTotal,donor[0],numbGifts,donorAverage])
+    
+    sortedReport = sorted(dataForDonationReport)
+    ascendingReport =sortedReport[::-1]
+    
+    for donor in ascendingReport:
+        donorTotal = "$ {0:.2f}".format(donor[0]) 
+        print('{:<20}'.format(donor[1]),'{:<15}'.format(donorTotal),
+                '{:>5}'.format(donor[2]),'{:>25}'.format(donor[3]))
+    getAction()                     
+
+
+if __name__ == "__main__" :
+  """   """
+donorList = [['Peter_Piper' ,100.00, 200.00 , 3000.00],['Lucille_Ball',100.00,200.00], ['Barak_Obama', 100.00,2000.00,3000.00],
+              ['Mary_Poppins', 1000, 2000.5], ['Bill_Gates', 100.0, 2000.0, 1000.0]]
+justDonors = ['Peter_Piper','Lucille_Ball', 'Barak_Obama', 'Mary_Poppins', 'Bill_Gates']
+
```

Comment:
Since you are not sending donorList and justDonors in as variables, they are necessary to have in the global (file) space, regardless of whether you run this as a script or import it, so they should go at the top, under the imports, if there are any. getAction is the only thing you need in the name/main block. Alternatively, you can start using input/output variables in your functions to pass around the data.